crates/evm/src/execute.rs

```diff
@@ -289,17 +311,220 @@ pub trait BlockExecutionStrategyFactory: ConfigureEvmFor<Self::Primitives> + 'st
         self.create_strategy(evm, ctx)
     }
 
+    /// Creates a [`BlockBuilder`] with the given [`BlockFactory`].
+    fn create_block_builder<'a, DB, I>(
```

Comment:
here we should also explain what the difference between strategy and builder really is, like when to use which